airflow/providers/amazon/aws/links/emr.py

```diff
@@ -66,3 +82,111 @@ def get_log_uri(
         return None
     log_uri = S3Hook.parse_s3_url(cluster_info["LogUri"])
     return "/".join(log_uri)
+
+
+class EmrServerlessLogsLink(BaseAwsLink):
+    """Helper class for constructing Amazon EMR Serverless link to Spark stdout logs."""
+
+    name = "Spark Driver stdout"
+    key = "emr_serverless_logs"
+
+    def get_link(
+        self,
+        operator: BaseOperator,
+        *,
+        ti_key: TaskInstanceKey,
+    ) -> str:
+        """
+        Pre-signed URL to the Spark stdout log.
+
+        :param operator: airflow operator
+        :param ti_key: TaskInstance ID to return link for
+        :return: Pre-signed URL to Spark stdout log. Empty string if no Spark stdout log is available.
+        """
+        conf = XCom.get_value(key=self.key, ti_key=ti_key)
+        if not conf:
+            return ""
+        # If get_dashboard_for_job_run fails for whatever reason, fail after 1 attempt
+        # so that the rest of the links load in a reasonable time frame.
+        hook = EmrServerlessHook(
+            aws_conn_id=conf.get("conn_id"), config={"retries": {"total_max_attempts": 1}}
+        )
+        resp = hook.conn.get_dashboard_for_job_run(
+            applicationId=conf.get("application_id"), jobRunId=conf.get("job_run_id")
+        )
+        o = urlparse(resp["url"])
+        return o._replace(path="/logs/SPARK_DRIVER/stdout.gz").geturl()
```

Comment:
I guess this parts could be moved into the `format_link` method rather than redefine entire method.
`get_link`. Original one suppress any errors (except subclasses of BaseException)

https://github.com/apache/airflow/blob/16ffc87ff42f702221f6cb7e42e08bc208183cf1/airflow/providers/amazon/aws/links/base_aws.py#L64-L65

Reply:
I think I put this in `get_link` as I wasn't making using of the `format_str` functionality in `format_link`. Fine with either, I think, although the benefit of suppressing the errors is that there are scenarios this call could fail and it would be ok. eg:
- User doesn't have access to that API
- Job is not in the right state
- Network failure

If the call fails, I'm fine with the link not returning.

Reply:
Suppressing here happen for different reason:
- Avoid to crash webserver in worst scenario
- One faulty link might cause that other wouldn't rendered

The error with traceback will be printed with debug severity anyway.

Reply:
Yeah `format_link` in general use for format from XCom configuration by apply to the predefined URI.
But there is no restriction to use it for custom implementation in case if you do not need anything else rather than XCom configuration.